sympy/printing/ccode.py

```diff
@@ -733,141 +736,3 @@ def _print_alignof(self, expr):
     'c99': C99CodePrinter,
     'c11': C11CodePrinter
 }
-
-
-def ccode(expr, assign_to=None, standard='c99', **settings):
```

Comment:
It seems weird to move this function - `latex` and `print_latex` are within `latex.py`, so keeping the convenience functions next to the printer subclasses looks to be the norm.

Reply:
I suppose the change here is that the convenience functions no longer import their dependencies until needed for the first time, which I guess improved import time.

Can we make all the printers consistent in this approach? Or can we make users just write the longer import to explicitly import `print_fcode` etc?

Reply:
Yes, specifically the sympy.codegen module does not need to be imported at all when sympy is imported. Unfortunately, the codegen objects are used in the class body, so there's no way to avoid importing them at the top-level of this file. So the next best thing is to just avoid importing this file. 

In general I agree that this isn't a great change in terms of code organization, but I think it's worth it for this case because it removes an entire otherwise optional submodule from the top-level import. 

> Can we make all the printers consistent in this approach?

I was originally going to move all the printers around like this, but I decided against it. It's better to have the printers in the same files if possible. It's just these three printers that make extensive use of the codegen submodule. 

> Or can we make users just write the longer import to explicitly import print_fcode etc?

Well if users use the shorter imports like `from sympy import print_fcode` or even `from sympy.printing import print_fcode`, then it won't matter what file `print_fcode` is in. So I actually think that users importing things from fully qualified modules is not good, because it limits how we can reorganize code.

Reply:
And in general, Python's import semantics are often at odds with good code organization. You often find yourself fighting them to get the organization you want, either because of circular import issues, or issues like this where you want to avoid importing a module you don't need to, or other things. Sometimes you can make some compromises and win, e.g., by moving imports inside of functions. Other times you just have to give in and accept that your code can't be organized perfectly. I'm not aware of a general solution that always lets you organize your code in the way that is most natural, and also avoid issues with imports.

Reply:
How viable is moving imports inside functions here?

Reply:
The objects are used at the class definition level https://github.com/sympy/sympy/pull/19908/files/01563a4d26ae825c909a4668bd5fdba9685ed618#diff-7b342526581a07bce690949be8806f6bR164-R197. So it would not be very easy without changing how the class works.

Reply:
That link anchor doesn't go anyway - can you link to the lines on master / some git sha?

Reply:
https://github.com/sympy/sympy/blob/8544f01468187b9b27d4fdb4671ce672d03622f9/sympy/printing/ccode.py#L161-L213